cpp/src/gcylon/sorting/sorting.cpp

```diff
@@ -0,0 +1,263 @@
+    sampled_table = cudf::gather(tv, clmn->view());
+
+    return cylon::Status::OK();
+}
+
+cylon::Status GetSplitPointIndices(const cudf::table_view &sort_columns_tv,
+                                   const cudf::table_view &split_points_tv,
+                                   const std::vector<cudf::order> &column_orders,
+                                   cudf::null_order null_ordering,
+                                   std::vector<int32_t> &split_point_indices) {
+
+    std::vector<cudf::null_order> column_null_orders(sort_columns_tv.num_columns(), null_ordering);
+
+    auto split_point_indices_clmn = cudf::lower_bound(sort_columns_tv,
+                                                      split_points_tv,
+                                                      column_orders,
+                                                      column_null_orders);
+
+    split_point_indices.resize(split_point_indices_clmn->size(), 0);
+    auto status = cudaMemcpy(split_point_indices.data(),
+                             split_point_indices_clmn->view().data<uint8_t>(),
+                             split_point_indices_clmn->size() * 4,
+                             cudaMemcpyDeviceToHost);
+    if (status != cudaSuccess) {
+        return cylon::Status(cylon::Code::ExecutionError, "cudaMemcpy failed");
+    }
+
+    return cylon::Status::OK();
+}
+
+cylon::Status gcylon::DistributedSort(const cudf::table_view &tv,
+                                      const std::vector<int32_t> & sort_column_indices,
+                                      const std::vector<cudf::order> &column_orders,
+                                      std::shared_ptr<cylon::CylonContext> ctx,
+                                      std::unique_ptr<cudf::table> &sorted_table,
+                                      bool nulls_after,
+                                      const int sort_root) {
+
+    if (sort_column_indices.size() > tv.num_columns()) {
+        return cylon::Status(cylon::Code::ValueError,
+                      "number of values in sort_column_indices can not larger than the number of columns");
+    } else if (sort_column_indices.size() != column_orders.size()) {
+        return cylon::Status(cylon::Code::ValueError,
+                             "sizes of sort_column_indices and column_orders must match");
+    }
+
+    // sample_count = number_of_workers * SAMPLING_RATIO
+    const int SAMPLING_RATIO = 2;
+
+    // get a table_view with sort_columns in the first k positions
+    // since cudf::sort method expects all columns listed as sort keys
+    // we perform sorting on this table, and revert back to the original at the end
+    std::vector<int32_t> column_indices_4s;
+    for (int i = 0; i < tv.num_columns(); ++i) {
+        if (std::find(sort_column_indices.begin(), sort_column_indices.end(), i) == sort_column_indices.end()) {
+            column_indices_4s.push_back(i);
+        }
+    }
```

Comment:
I didnt quite understand this! but from the outset it simplifies without `std::find` like this, 
`if(sort_column_indices[sort_column_indices.size() -1] == i)`
Otherwise this becomes a O(N* num_columns) loop, isnt it?

Reply:
let's suppose the table has 6 columns and we would like to sort on columns(1,4):
sort_column_indices = [1,4]
we need to create column_indices_4s as:
column_indices_4s = [1,4,0,2,3,5]

so, we first insert sort_column_indices at the beginning (i changed it now to insert first to the beginning), 
then we push back indices that did not exist in sort_column_indices.